core/src/main/java/org/jsmart/zerocode/core/kafka/helper/KafkaConsumerHelper.java

```diff
@@ -79,20 +79,24 @@ public static Consumer createConsumer(String bootStrapServers, String consumerPr
         }
     }
 
-    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer) {
-        for (int run = 0; run < 10; run++) {
-            if (!consumer.assignment().isEmpty()) {
-                return new ConsumerRecords(new HashMap());
-            }
-            ConsumerRecords records = consumer.poll(Duration.of(500, ChronoUnit.MILLIS));
-            if (!records.isEmpty()) {
-                return records;
+    public static ConsumerRecords initialPollWaitingForConsumerGroupJoin(Consumer consumer, ConsumerLocalConfigs effectiveLocalConfigs) {
+
+            Long durationForPolling = ofNullable(effectiveLocalConfigs.getPollingTime()).orElse(500L);
```

Comment:
Why do you need the ofNullable check here, if I understand it correctly getPollingTime already provides null safety and returns the default polling times as defined in KafkaConstants?

Reply:
@ddingel i notice that effectiveLocalConfigs.getPollingTime() is returning NULL when its not set in properties. Sorry i should have tested this out early on without assuming. Hence you could see i explicitly set the default to 500L and believe would not require the number of runs increase you suggested.
Long durationForPolling = ofNullable(effectiveLocalConfigs.getPollingTime()).orElse(500L);

Reply:
Hi @ddingel,
>@authorjapps do I miss understand the current implementation, I thought the default now will be taken from KafkaConstants and will be 100ms?

if e.g. `consumer.pollingTime = 1000` is provided, then it will use this(1000), otherise it defaults to `500 mili sec` which a constant hardcoded.

This change also means:
`consumer.pollingTime = 1000 or 100` if provided, will be used for both purpose.
1) initialPollWaitingForConsumerGroupJoin = 1000 or 100 milisec, retries 10 times
2) Subsequent pollingTime = 1000 or 100 milisec(configurable as usual)

> DEFAULT_POLLING_TIME_MILLI_SEC (as @sparrowV wrote )

@dandalavinod, can you use this please, this should help @ddingel and backward compatibility is retained upto some extent.
> org.jsmart.zerocode.core.kafka.KafkaConstants#DEFAULT_POLLING_TIME_MILLI_SEC


Without this changes/earlier implementation:
1) initialPollWaitingForConsumerGroupJoin = 500 milisec HARD CODED, retries 10 times
   1.1)This is blocking other users like @dandalavinod as he needs slightly more time to rejoin e.g. 1000milisec
2) Subsequent pollingTime = configurable(100, 500, 1000, 5000 etc...)

***

Reply:
TLDR:
With getPollTime we use the correct default in all places and allow customisation. By increasing the number of runs we ensure that we do not change the default behaviour regarding kafka join, so that no user find themselves after doing an update with the new "Failed to join" messages in their tests.

---

Ahhh @dandalavinod, I think this solves the puzzle as @sparrowV pointed out the method to use is probably best:
```
    public static Long getPollTime(ConsumerLocalConfigs effectiveLocal) {
        return ofNullable(effectiveLocal.getPollingTime())
                .orElse(DEFAULT_POLLING_TIME_MILLI_SEC);
    }
```

This way we can ensure if at some point we change `DEFAULT_POLLING_TIME_MILLI_SEC` we get the initial polling as well :).


Hi @authorjapps, that the change will chain both constants and unify the handling to use a single value is a good thing. Users will be able to adjust. I was just confused where the `500 milisec` came from. As this is the previous hardcoded value :).

Reply:
Done - updated code to reuse getPollTime function